app/jobs/csv_export_job.rb

```diff
@@ -0,0 +1,72 @@
+require 'csv'
+
+class CsvExportJob < ActiveJob::Base
+  queue_as :csv_jobs
+ 
+  def perform(csv_export_id)
+    ActiveRecord::Base.connection_pool.with_connection do
+      csv_export = CsvExport.find(csv_export_id)
```

Comment:
I'm pretty sure ActiveJob can actually do this for you - can't you just pass in the model rather than its id?